nimare/meta/cbma/base.py

```diff
@@ -0,0 +1,476 @@
+        """
+        super()._preprocess_input(dataset)
+
+        # All extra (non-ijk) parameters for a kernel should be overrideable as
+        # parameters to __init__, so we can access them with get_params()
+        kt_args = list(self.kernel_transformer.get_params().keys())
+
+        # Integrate "sample_size" from metadata into DataFrame so that
+        # kernel_transformer can access it.
+        if "sample_size" in kt_args:
+            if "sample_sizes" in dataset.get_metadata():
+                # Extract sample sizes and make DataFrame
+                sample_sizes = dataset.get_metadata(field="sample_sizes", ids=dataset.ids)
+                # we need an extra layer of lists
+                sample_sizes = [[ss] for ss in sample_sizes]
+                sample_sizes = pd.DataFrame(
+                    index=dataset.ids, data=sample_sizes, columns=["sample_sizes"]
+                )
+                sample_sizes["sample_size"] = sample_sizes["sample_sizes"].apply(np.mean)
+                # Merge sample sizes df into coordinates df
+                self.inputs_["coordinates"] = self.inputs_["coordinates"].merge(
+                    right=sample_sizes,
+                    left_on="id",
+                    right_index=True,
+                    sort=False,
+                    validate="many_to_one",
+                    suffixes=(False, False),
+                    how="left",
+                )
+            else:
+                LGR.warning(
+                    'Metadata field "sample_sizes" not found. '
+                    "Set a constant sample size as a kernel transformer "
+                    "argument, if possible."
+                )
+
+    def compute_summarystat(self, data):
+        """Compute OF scores from data.
+
+        Parameters
+        ----------
+        data : array, pandas.DataFrame, or list of img_like
+            Data from which to estimate ALE scores.
+            The data can be:
+            (1) a 1d contrast-len or 2d contrast-by-voxel array of MA values,
+            (2) a DataFrame containing coordinates to produce MA values,
+            or (3) a list of imgs containing MA values.
+
+        Returns
+        -------
+        stat_values : 1d array
+            OF values. One value per voxel.
+        """
+        if isinstance(data, pd.DataFrame):
+            ma_values = self.kernel_transformer.transform(
+                data, masker=self.masker, return_type="array"
+            )
+        elif isinstance(data, list):
+            ma_values = self.masker.transform(data)
+        elif isinstance(data, np.ndarray):
+            ma_values = data
+        elif not isinstance(data, np.ndarray):
+            raise ValueError('Unsupported data type "{}"'.format(type(data)))
+
+        # Apply weights before returning
+        return self._compute_summarystat(ma_values)
+
+    def _compute_summarystat(self, ma_values):
+        pass
```

Comment:
Should this be an abstract method?

Reply:
It will be, but only once we've refactored `MKDAChi2` to have the same general architecture, which I'm planning to do separately. Otherwise it will break that class. (I suppose we could define a placeholder method in `MKDAChi2` if you like.)